### DISCUSSION TITLE
Compiling OpenThread with no CLI

### DISCUSSION THREAD

**Discussion by 10andy0** [openthread/openthread#9838] [Q&A] [answered]
Hi there,
    I am trying to reduce size of the binary (MTD). Was wondering if it is possible to compile without CLI support?
Cheers,
Andy

**Comment by jwhui** [comment, marked as answer]
Yes, OT CLI is compiled as a separate library.

https://github.com/openthread/openthread/blob/905a22e0c7ea2a50fd3b935c24eb200f83721455/examples/apps/cli/mtd.cmake#L40

Simply do not link or reference any CLI-related functions.
